kernel/src/hil/flash.rs

```diff
@@ -111,10 +111,18 @@ pub trait Flash {
     type Page: AsMut<[u8]>;
 
     /// Read a page of flash into the buffer.
-    fn read_page(&self, page_number: usize, buf: &'static mut Self::Page) -> ReturnCode;
+    fn read_page(
+        &self,
+        page_number: usize,
+        buf: &'static mut Self::Page,
+    ) -> (ReturnCode, Option<&'static mut Self::Page>);
 
     /// Write a page of flash from the buffer.
-    fn write_page(&self, page_number: usize, buf: &'static mut Self::Page) -> ReturnCode;
+    fn write_page(
+        &self,
+        page_number: usize,
+        buf: &'static mut Self::Page,
+    ) -> (ReturnCode, Option<&'static mut Self::Page>);
```

Comment:
These seem like good changes to the HIL, but you only modified the sam4l implementation of the HIL, so know all other chips will fail to compile (which is why Travis is failing). I think it makes sense to break out the changes to the HIL / / chip-specific flash drivers / non-log capsules into its own PR, which can be reviewed and merged independently of the log capsules and tests for the log capsule.